filter_k8s_servicetypes.go

```diff
@@ -0,0 +1,32 @@
+package aiven
+
+var _ Filter = (*k8sServiceTypesFilter)(nil)
+
+// k8sSupportedServiceTypes is a map of service types supported in Kubernetes operator.
+var k8sSupportedServiceTypes = map[string]struct{}{
```

Comment:
Perhaps it is not the best idea to have it as a global package variable, because any user can accidentally add or remove something from map. 

The better option would be: `func k8sSupportedServiceTypes() []string`

Also, why do we need a map of empty structures where we only care about the key? Can we change it to the strings slice?

Reply:
> Perhaps it is not the best idea to have it as a global package variable, because any user can accidentally add or remove something from map.

it is not a global variable, it's a local package variable and shouldn't be accessible from outside this project
I moved it to another file, though

> The better option would be: `func k8sSupportedServiceTypes() []string`

the better option isn't available in Go at the moment, which is const slices :(

> Also, why do we need a map of empty structures where we only care about the key? Can we change it to the strings slice?

it saves a lot of CPU cycles to filter it this way, there are more efficient loops in the assembled code and it uses the hash table search algorithm instead of a linear search, and we don't even need to maintain the search algorithm :)

Reply:
this is no longer relevant